sympy/functions/combinatorial/factorials.py

```diff
@@ -339,13 +339,18 @@ def eval(cls, arg):
                 if arg.is_even:
                     k = arg / 2
                     return 2 ** k * factorial(k)
-
-                return factorial(arg) / factorial2(arg - 1)
+                if arg.is_odd:
+                    return factorial(arg) / factorial2(arg - 1)
+                raise ValueError("argument must be nonnegative integer or odd integer")
 
             if arg.is_even:
                 raise ValueError("argument must be nonnegative or odd")
 
-            return arg * (S.NegativeOne) ** ((1 - arg) / 2) / factorial2(-arg)
+            if arg.is_integer:
```

Comment:
@jksuom Thank you for your help. But the thing is that, We can't replace arg.is_infinite with either of not arg.is_Integer or not arg.is_integer , because , if you initialize a symbol 'x' with no other additional parameters, and no values assigned as well, then always not arg.is_Integer or not arg.is_integer be 'true' and hence it'll enter the first loop, and raise a VaueError instead of returning None. This was the same problem happening in my previous commit, i.e, the commit previous to this one, because I was raising a ValueError in the last 'else' statement, where I am returning None now. That was causing a lot of tests to fail in the Travis CI build. Although yes, as you told the last 'else' statement is not required, I agree to that. Python will itself return none, if nothing is specified. 

So, I feel we can't replace arg.is_infinite with the other two. So, could you help me out as to why the test is failing in the Travis CI Build. There's only one test which is failing. I am not able to figure out, as to why it's failing. :( 
@gxyd @Shekharrajak @asmeurer